### ISSUE TITLE
Inquiry about lit-llama's training speed

### ISSUE THREAD

**Issue by tshu-w** [Lightning-AI/lit-llama#119]
It would be helpful to know how fast lit-llama can be trained as it is crucial for pre-training costs. Some comparable data can be found in the link provided: [https://github.com/s-JoL/Open-Llama#和其他开源模型性能对比](https://github.com/s-JoL/Open-Llama#%E5%92%8C%E5%85%B6%E4%BB%96%E5%BC%80%E6%BA%90%E6%A8%A1%E5%9E%8B%E6%80%A7%E8%83%BD%E5%AF%B9%E6%AF%94).

**Comment by lantiga** [comment]
We are starting looking into pre-training now, so giving you a number might be premature, but definitely thanks for sharing this. We'll be posting numbers as soon as we are ready.

**Comment by lantiga** [comment]
First timings from #217, with that we are around 2k tokens/sec/GPU
- no xformers
- no torch.compile (unfortunately activation checkpointing is not supported by torch.compile yet, and without it we OOM with 40GB)

**Comment by tshu-w** [comment]
> It would be interesting to understand what cloud help closing the gaps. What machine did you use to run your benchmarks?

Thank you for providing the information. During training on the internal server (A100 80Gx8), approximately ~2.9k tokens/sec/GPU were achieved using large gradient accumulation and Magatron-Deepspeed.

The link above is a project I found, but it's unclear where it was trained on cloud.

**Comment by lantiga** [comment]
Yeah I think the extra memory (40GB vs 80GB) allows for a larger micro batch size and can really give you an edge there.
We'll test it in the future, in the meantime I'm running further tests and will report back.

**Comment by lantiga** [comment]
As an update, I'm getting 2.2k by using a larger effective batch size during gradient accumulation (since it amortizes the optimizer step).
Feel that moving to a larger micro batch on a 80GB node and using torch.compile (or something similar) to fuse ops will fill the gap substantially.
